sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/FailureSafeParser.scala

```diff
@@ -67,6 +86,8 @@ class FailureSafeParser[IN](
         case FailFastMode =>
           throw QueryExecutionErrors.malformedRecordsDetectedInRecordParsingError(e)
       }
+      case _ if (mode == DropMalformedMode) =>
```

Comment:
we should only match `IllegalSchemaArgumentException`. We should still fail for `SparkUpgradeException `